components/button/src/Button.js

```diff
@@ -0,0 +1,276 @@
+/* @jsx jsx */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { jsx, useTheme } from '@westpac/core';
+import { mediaQueries } from './utils';
+
+// ==============================
+// Utils
+// ==============================
+
+const getSizeStyling = (size, trim, button) => {
+	const padding = button.size[size].padding;
+	if (trim) {
+		padding[1] = 0;
+	}
+
+	return {
+		padding: padding.join(' '), //provided as an array
+		fontSize: button.size[size].fontSize,
+		height: button.size[size].height,
+	};
+};
+
+const getBlockStyling = block => {
+	return block
+		? // Block mode
+		  {
+				display: 'flex', //flex will provide 'block-level' appearance (we use flex so any icon children can be positioned appropriately)
+				width: '100%',
+		  }
+		: // Inline mode
+		  {
+				display: 'inline-flex',
+				width: 'auto', //reset
+		  };
+};
+
+// ==============================
+// Component
+// ==============================
+
+export const Button = ({
+	appearance,
+	size,
+	soft,
+	block,
+	trim,
+	icon: Icon,
+	iconPosition,
+	justify,
+	children,
+	tag: Tag,
+	onClick,
+	...props
+}) => {
+	const theme = useTheme();
+	const mq = mediaQueries(theme.breakpoints);
```

Comment:
replace with common paint util

Reply:
Will we always need to `@emotion/weak-memoize` or only for grid?